# บทนำ: การใช้งาน Jupyter Notebook เบื้องต้น

ในหลักสูตรนี้ เราจะใช้ **Jupyter Notebook** เป็นเครื่องมือหลักสำหรับเรียนรู้ Python และการวิเคราะห์ข้อมูล

Jupyter Notebook เป็นพื้นที่ทำงานที่ช่วยให้เราสามารถ

- เขียนโค้ด Python
- รันโค้ดทีละส่วน
- ดูผลลัพธ์ทันที
- จดบันทึก อธิบายแนวคิด และสรุปผลการวิเคราะห์
- ทำงานวิเคราะห์ข้อมูลในรูปแบบที่ตรวจสอบย้อนหลังได้

สำหรับผู้ที่คุ้นเคยกับ Excel ให้มองว่า Notebook คล้ายกับ “สมุดวิเคราะห์ข้อมูล” ที่มีทั้งคำอธิบาย ขั้นตอนการทำงาน และผลลัพธ์อยู่ในไฟล์เดียวกัน

## ผลการเรียนรู้ที่คาดหวัง

เมื่อเรียนจบบทนี้ ผู้เรียนจะสามารถ

1. อธิบายบทบาทของ Jupyter Notebook ในงานวิเคราะห์ข้อมูลได้
2. แยกความแตกต่างระหว่าง Code Cell และ Markdown Cell ได้
3. รันคำสั่ง Python ใน Code Cell และอ่านผลลัพธ์เบื้องต้นได้
4. ใช้ Markdown Cell เพื่อเขียนคำอธิบายหรือบันทึกผลการวิเคราะห์ได้
5. เข้าใจลำดับการทำงานแบบ เขียน → รัน → ตรวจสอบผลลัพธ์
6. ใช้ Python ทำงานพื้นฐานบางอย่างที่เคยทำใน Excel ได้ เช่น การคำนวณสูตรง่าย ๆ การหาผลรวม และการหาค่าเฉลี่ย

## ลำดับเนื้อหา

บทเรียนนี้ประกอบด้วยหัวข้อต่อไปนี้

1. Jupyter Notebook คืออะไร
2. ส่วนประกอบพื้นฐานของ Notebook
3. Code Cell คืออะไร
4. Markdown Cell คืออะไร
5. วิธีการรัน Cell
6. การใช้ Python แทนสูตรพื้นฐานใน Excel
7. การทำงานกับข้อมูลหลายค่า
8. ลำดับการทำงานใน Notebook
9. ข้อควรระวังเบื้องต้นในการใช้ Notebook
10. ตัวอย่าง Error เบื้องต้น
11. การเขียนบันทึกผลการวิเคราะห์
12. แบบฝึกหัดท้ายบท

## 1. Jupyter Notebook คืออะไร

**Jupyter Notebook** คือเครื่องมือที่ช่วยให้เราเขียนและรันโค้ด Python แบบทีละส่วนได้

จุดเด่นของ Jupyter Notebook คือ เราสามารถรวมสิ่งต่อไปนี้ไว้ในที่เดียวกัน

- โค้ด Python
- ผลลัพธ์จากการรันโค้ด
- คำอธิบาย
- ตาราง
- กราฟ
- บันทึกการวิเคราะห์

ในงานวิเคราะห์ข้อมูล Notebook ช่วยให้เราทำงานอย่างเป็นขั้นตอน และสามารถย้อนกลับมาตรวจสอบได้ว่าเราใช้ข้อมูลอย่างไร คำนวณอย่างไร และได้ข้อสรุปอย่างไร

## เปรียบเทียบ Jupyter Notebook กับ Excel

| งานที่ทำ | Excel | Jupyter Notebook / Python |
|---|---|---|
| ใส่สูตรคำนวณ | พิมพ์สูตรใน Cell เช่น `=A1+B1` | เขียนคำสั่ง Python เช่น `100 + 200` |
| ดูผลลัพธ์ | แสดงผลใน Cell | แสดงผลใต้ Code Cell |
| จดคำอธิบาย | ใช้ Cell หรือ Comment | ใช้ Markdown Cell |
| ทำงานซ้ำ | Copy สูตร / Drag สูตร | เขียนโค้ดให้ทำซ้ำ |
| วิเคราะห์ข้อมูล | Filter, Sort, PivotTable | ใช้ Python และ pandas |
| ตรวจสอบขั้นตอนย้อนหลัง | ดูสูตรในแต่ละ Cell | อ่าน Notebook จากบนลงล่าง |

## 2. ส่วนประกอบพื้นฐานของ Notebook

Notebook ประกอบด้วยหน่วยย่อยที่เรียกว่า **Cell**

Cell คือพื้นที่แต่ละช่องใน Notebook ที่เราใช้เขียนเนื้อหา

Cell ที่ใช้บ่อยมี 2 ประเภทหลัก

1. **Code Cell**  
   ใช้เขียนและรันโค้ด Python

2. **Markdown Cell**  
   ใช้เขียนข้อความ คำอธิบาย หัวข้อ ตาราง หรือบันทึกประกอบการวิเคราะห์

## 3. Code Cell คืออะไร

**Code Cell** คือ Cell ที่ใช้เขียนคำสั่ง Python

เมื่อรัน Code Cell แล้ว ผลลัพธ์จะแสดงอยู่ด้านล่างของ Cell นั้น

In [23]:
print("Hello, Jupyter Notebook")

Hello, Jupyter Notebook


จากตัวอย่างด้านบน

- `print()` คือคำสั่งที่ใช้แสดงผล
- ข้อความที่อยู่ในเครื่องหมาย `" "` คือข้อความที่ต้องการให้แสดงออกมา
- เมื่อรัน Cell แล้ว ผลลัพธ์จะแสดงด้านล่าง

In [24]:
print("สวัสดีครับ/ค่ะ")
print("เรากำลังเริ่มเรียน Python สำหรับงานวิเคราะห์ข้อมูล")

สวัสดีครับ/ค่ะ
เรากำลังเริ่มเรียน Python สำหรับงานวิเคราะห์ข้อมูล


## 4. Markdown Cell คืออะไร

**Markdown Cell** คือ Cell ที่ใช้เขียนข้อความอธิบาย ไม่ได้ใช้รันโค้ด Python

เราสามารถใช้ Markdown Cell เพื่อ

- เขียนหัวข้อ
- อธิบายขั้นตอนการวิเคราะห์
- บันทึกสมมติฐาน
- สรุปผลลัพธ์
- ทำตารางประกอบคำอธิบาย

ในการทำงานจริง Markdown Cell มีความสำคัญมาก เพราะช่วยให้ผู้อื่น รวมถึงตัวเราในอนาคต เข้าใจว่า Notebook นี้ทำอะไร

### ตัวอย่าง Markdown

เราสามารถเขียนรายการเป็น bullet ได้ เช่น

- ขั้นตอนที่ 1: โหลดข้อมูล
- ขั้นตอนที่ 2: ตรวจสอบข้อมูล
- ขั้นตอนที่ 3: วิเคราะห์ข้อมูล
- ขั้นตอนที่ 4: สรุปผล

หรือเขียนตัวหนา เช่น **ข้อความสำคัญ**

หรือเขียน code สั้น ๆ ในบรรทัดเดียว เช่น `print("Hello")`

## 5. วิธีการรัน Cell

การรัน Cell คือการสั่งให้ Notebook ประมวลผล Cell นั้น

วิธีที่ใช้บ่อย:

- กดปุ่ม **Run**
- หรือกดแป้นพิมพ์ **Shift + Enter**

เมื่อรัน Code Cell แล้ว Notebook จะแสดงผลลัพธ์อยู่ด้านล่าง Cell

ถ้าเป็น Markdown Cell เมื่อรันแล้ว ข้อความจะถูกจัดรูปแบบให้อ่านง่ายขึ้น

In [25]:
10 + 20

30

ใน Code Cell ด้านบน เราใช้ Python เป็นเหมือนเครื่องคิดเลข

ใน Excel เราอาจเขียนว่า

```excel
=10+20
```

ใน Python เราเขียนว่า

```python
10 + 20
```

## 6. การใช้ Python แทนสูตรพื้นฐานใน Excel

ก่อนที่เราจะเรียน Python อย่างเป็นระบบในบทถัดไป เราจะทดลองใช้ Python ทำงานง่าย ๆ ที่คล้ายกับการใช้สูตรใน Excel

### ตัวอย่างที่ 1: การบวกเลข

ใน Excel:

```excel
=100+250
```

ใน Python:

In [26]:
100 + 250

350

### ตัวอย่างที่ 2: การคำนวณยอดขายรวม

สมมติว่าเรามีข้อมูลดังนี้

| รายการ | จำนวน | ราคาต่อหน่วย |
|---|---:|---:|
| สินค้า A | 10 | 120 |
| สินค้า B | 5 | 300 |

ใน Excel เราอาจคำนวณยอดขายด้วยสูตร

```excel
=จำนวน*ราคาต่อหน่วย
```

ใน Python เราสามารถเขียนได้ดังนี้

In [27]:
quantity_a = 10
price_a = 120

sales_a = quantity_a * price_a

sales_a

1200

In [28]:
quantity_b = 5
price_b = 300

sales_b = quantity_b * price_b

sales_b

1500

In [29]:
total_sales = sales_a + sales_b

total_sales

2700

จากตัวอย่างนี้ เราเริ่มเห็นแนวคิดสำคัญของ Python คือ

- เราสามารถตั้งชื่อให้ข้อมูล เช่น `quantity_a`
- เราสามารถนำข้อมูลมาคำนวณได้
- เราสามารถเก็บผลลัพธ์ไว้ใช้ต่อได้ เช่น `sales_a`, `sales_b`, `total_sales`

แนวคิดนี้คล้ายกับการอ้างอิง Cell ใน Excel แต่ Python จะใช้ “ชื่อตัวแปร” แทนตำแหน่ง Cell

## 7. การทำงานกับข้อมูลหลายค่า

ใน Excel เรามักมีข้อมูลหลายแถว เช่น ยอดขายหลายรายการ

ใน Python เราสามารถเก็บข้อมูลหลายค่าไว้ในรูปแบบที่เรียกว่า **list**

ยังไม่ต้องกังวลเรื่องรายละเอียดของ list ในตอนนี้ เพราะเราจะเรียนอย่างเป็นระบบในบทถัดไป

ในบทนี้ให้เข้าใจก่อนว่า list คือการเก็บค่าหลายค่าไว้ด้วยกัน

In [30]:
sales = [1200, 1500, 800, 2200, 1750]

sales

[1200, 1500, 800, 2200, 1750]

### การหาผลรวม

ใน Excel เราใช้

```excel
=SUM(A1:A5)
```

ใน Python เราใช้ sum()

In [31]:
sum(sales)

7450

### การหาจำนวนรายการ

ใน Excel เราอาจใช้

```excel
=COUNT(A1:A5)
```

ใน Python เราใช้ len()

In [32]:
len(sales)

5

### การหาค่าเฉลี่ย

ใน Excel เราใช้

```excel
=AVERAGE(A1:A5)
```

ใน Python เราสามารถคำนวณจากผลรวม หารด้วยจำนวนรายการ

In [33]:
average_sales = sum(sales) / len(sales)

average_sales

1490.0

### การหาค่าสูงสุดและต่ำสุด

ใน Excel เราใช้

```excel
=MAX(A1:A5)
=MIN(A1:A5)
```

ใน Python เราใช้ max() และ min()

In [34]:
max(sales)

2200

In [35]:
min(sales)

800

## 8. ลำดับการทำงานใน Notebook

การทำงานใน Notebook ควรอ่านจากบนลงล่าง

โดยทั่วไปเราจะทำงานตามลำดับนี้

1. อธิบายวัตถุประสงค์
2. เตรียมข้อมูลหรือกำหนดค่าเริ่มต้น
3. เขียนโค้ดเพื่อคำนวณหรือวิเคราะห์
4. ตรวจสอบผลลัพธ์
5. เขียนสรุปหรือข้อสังเกต

แนวทางนี้จะช่วยให้ Notebook ไม่ใช่แค่ที่เขียนโค้ด แต่เป็นเอกสารการวิเคราะห์ข้อมูลที่อ่านเข้าใจได้

## 9. ข้อควรระวังเบื้องต้นในการใช้ Notebook

### 1. ลำดับการรัน Cell มีผลต่อผลลัพธ์

ถ้าเรารัน Cell ด้านล่างก่อน Cell ด้านบน อาจเกิด error ได้ เพราะ Python ยังไม่รู้จักค่าที่เราต้องการใช้

### 2. ถ้าแก้ไขค่า ต้องรัน Cell ใหม่

ถ้าเปลี่ยนข้อมูลใน Cell หนึ่ง ควรรัน Cell ที่เกี่ยวข้องอีกครั้ง

### 3. Kernel คือหน่วยความจำของ Notebook

Kernel เป็นเหมือนพื้นที่ที่ Python ใช้จำค่าต่าง ๆ ที่เราสร้างไว้

ถ้า Restart Kernel ค่าที่เคยเก็บไว้จะหายไป และต้องรัน Cell ใหม่ตั้งแต่ต้น

## 10. ตัวอย่าง Error เบื้องต้น

Error เป็นเรื่องปกติในการเขียน Python

เวลาเจอ Error ไม่ต้องตกใจ ให้เริ่มจากอ่านบรรทัดสุดท้ายก่อน เพราะมักบอกสาเหตุสำคัญที่สุด

In [36]:
# ลองรัน Cell นี้ แล้วสังเกต Error ที่เกิดขึ้น

total_revenue

5300

Code ด้านบนอาจทำให้เกิด Error เพราะเรายังไม่ได้สร้างตัวแปรชื่อ `total_revenue`

Error ลักษณะนี้มักเกิดจาก

- พิมพ์ชื่อตัวแปรผิด
- ยังไม่ได้รัน Cell ที่สร้างตัวแปรนั้น
- Restart Kernel แล้วลืมรัน Cell ก่อนหน้าใหม่

ในหลักสูตรนี้ เราจะค่อย ๆ ฝึกอ่าน Error ไปพร้อมกัน

In [37]:
# แก้ไขโดยสร้างตัวแปรก่อน

total_revenue = 1000 + 2500 + 1800

total_revenue

5300

## 11. การเขียนบันทึกผลการวิเคราะห์

ในการวิเคราะห์ข้อมูล ไม่ใช่แค่การคำนวณให้ได้คำตอบ แต่ต้องสามารถอธิบายได้ว่า

- ข้อมูลที่ใช้คืออะไร
- คำนวณอย่างไร
- ได้ผลลัพธ์อะไร
- ผลลัพธ์นั้นหมายความว่าอะไร

Markdown Cell จึงมีบทบาทสำคัญมาก

### ตัวอย่างบันทึกผลการวิเคราะห์

จากข้อมูลยอดขาย 5 รายการ พบว่า

- ยอดขายรวมเท่ากับ 7,450 บาท
- ยอดขายเฉลี่ยเท่ากับ 1,490 บาท
- ยอดขายสูงสุดเท่ากับ 2,200 บาท
- ยอดขายต่ำสุดเท่ากับ 800 บาท

ข้อสังเกตเบื้องต้นคือ ยอดขายแต่ละรายการมีความแตกต่างกันพอสมควร และควรตรวจสอบต่อว่าแต่ละยอดขายมาจากสินค้า ประเภทลูกค้า หรือช่วงเวลาใด

# แบบฝึกหัดท้ายบท  

ในแบบฝึกหัดนี้ ให้ผู้เรียนทดลองเปลี่ยนงานที่เคยทำใน Excel มาใช้ Python แทน

ยังไม่จำเป็นต้องเข้าใจ Python ทุกส่วนอย่างละเอียด จุดประสงค์ของแบบฝึกหัดนี้คือให้คุ้นเคยกับการใช้ Notebook และการรัน Code Cell

## แบบฝึกหัดที่ 1: ใช้ Python แทนเครื่องคิดเลขใน Excel

ใน Excel เราอาจเขียนสูตร

```excel
=1500+2300+1800
```

ให้เขียน Python เพื่อคำนวณผลรวมของตัวเลขทั้งสามจำนวน

In [38]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 2: คำนวณยอดขาย

สมมติว่ามีข้อมูลดังนี้

| สินค้า | จำนวนขาย | ราคาต่อหน่วย |
|---|---:|---:|
| Product A | 25 | 120 |
| Product B | 18 | 250 |
| Product C | 30 | 90 |

ใน Excel เราอาจคำนวณยอดขายของแต่ละสินค้าโดยใช้สูตร

```excel
=จำนวนขาย*ราคาต่อหน่วย
```

ให้ใช้ Python คำนวณ

ยอดขายของ Product A
ยอดขายของ Product B
ยอดขายของ Product C
ยอดขายรวมทั้งหมด

In [39]:
# กำหนดจำนวนขายและราคาต่อหน่วย

quantity_a = 25
price_a = 120

quantity_b = 18
price_b = 250

quantity_c = 30
price_c = 90

In [40]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 3: หาผลรวม ค่าเฉลี่ย ค่าสูงสุด และค่าต่ำสุด

ใน Excel เราอาจใช้สูตร

```excel
=SUM(A1:A6)
=AVERAGE(A1:A6)
=MAX(A1:A6)
=MIN(A1:A6)
```

ให้ใช้ Python เพื่อหาค่าต่อไปนี้จากข้อมูลยอดขายด้านล่าง

```python
monthly_sales = [12500, 18300, 15700, 22100, 19800, 17600]
```

ให้คำนวณ

* ยอดขายรวม
* ยอดขายเฉลี่ย
* ยอดขายสูงสุด
* ยอดขายต่ำสุด

In [41]:
monthly_sales = [12500, 18300, 15700, 22100, 19800, 17600]

In [42]:
# เขียนคำตอบของคุณใน Cell นี้

## แบบฝึกหัดที่ 4: เขียนบันทึกผลการวิเคราะห์ด้วย Markdown

จากผลลัพธ์ในแบบฝึกหัดที่ 3 ให้เพิ่ม Markdown Cell ด้านล่าง แล้วเขียนสรุปผลการวิเคราะห์สั้น ๆ

ตัวอย่างรูปแบบคำตอบ

```markdown
จากข้อมูลยอดขายรายเดือน พบว่า

- ยอดขายรวมเท่ากับ ...
- ยอดขายเฉลี่ยเท่ากับ ...
- ยอดขายสูงสุดเท่ากับ ...
- ยอดขายต่ำสุดเท่ากับ ...

ข้อสังเกตเบื้องต้นคือ ...

## แบบฝึกหัดที่ 5: ทดลองแก้ไขและรัน Cell ใหม่

ให้ทดลองทำตามขั้นตอนต่อไปนี้

1. กลับไปที่ Cell ซึ่งกำหนดค่า `monthly_sales`
2. เพิ่มยอดขายเดือนที่ 7 เข้าไป เช่น `20500`
3. รัน Cell นั้นใหม่
4. รัน Cell ที่ใช้คำนวณผลรวม ค่าเฉลี่ย ค่าสูงสุด และค่าต่ำสุดอีกครั้ง
5. สังเกตว่าผลลัพธ์เปลี่ยนไปอย่างไร

คำถาม:

- ผลรวมเปลี่ยนไปหรือไม่
- ค่าเฉลี่ยเปลี่ยนไปหรือไม่
- ค่าสูงสุดเปลี่ยนไปหรือไม่
- สิ่งนี้คล้ายกับการแก้ข้อมูลใน Excel แล้วสูตรคำนวณใหม่อย่างไร